### PR TITLE
CRM-20910: Check permission param while retrieving participants from api

### PR DESCRIPTION
**Pull request by jitendrapurohit**
Unit test added.

* [CRM-20910: API call to Participant.get ignores check_permissons](https://issues.civicrm.org/jira/browse/CRM-20910)

### CODE REVIEW

**Comment by jitendrapurohit** [comment]
jenkins, retest this please

**Comment by monishdeb** [comment]
Jenkins test this please

File: CRM/Event/BAO/Query.php [hunk 1]
```diff
@@ -253,6 +253,10 @@ public static function where(&$query) {
    * @param $query
    */
   public static function whereClauseSingle(&$values, &$query) {
+    $checkPermission = TRUE;
+    if (!empty($query->_skipPermission)) {
+      $checkPermission = FALSE;
+    }
```
Comment (colemanw) [inline CRM/Event/BAO/Query.php]: This could be written in 1 line: 

    $checkPermission = empty($query->_skipPermission);

**Review by colemanw — CHANGES_REQUESTED** [review]

File: CRM/Event/BAO/Query.php (hunk 1 above)
Comment (jitendrapurohit) [inline CRM/Event/BAO/Query.php]: Updated 👍

**Comment by jitendrapurohit** [comment]
How do I change the status of this PR as I've updated the changed that were requested.

**Review by colemanw — APPROVED** [review]

**Pull request merged**